starlark/eval.go

```diff
@@ -84,6 +84,12 @@ func (thread *Thread) SetMaxExecutionSteps(max uint64) {
 	thread.maxSteps = max
 }
 
+// ResetStepsAndCancelReason reset steps counts and cancel reason
+func (thread *Thread) ResetStepsAndCancelReason() {
+	thread.steps = 0
```

Comment:
If we're going to allow users to reset it, then I don't really see any reason not to make the steps variable public; let's do that, and move the comment from ExecutionSteps to it. ExecutionSteps (which we must retain for compatibility) can just say "returns the current value of Steps".

Then this function need only reset the cancellation state. Call it Uncancel, and update the comment "Cancellation cannot be undone" to read "Call [Uncancel] to reset the cancellation state".  For symmetry, this method should use atomic.SetPointer and copy the "Unlike most methods..." comment from Cancel.

(The fact that we use the "reason" *string as the representation is an implementation detail that doesn't belong in the method name.)

Reply:
Based on your suggestion, I submitted some changes